crates/vault-registry/src/lib.rs

```diff
@@ -1588,22 +1659,56 @@ impl<T: Config> Pallet<T> {
+
+        let fee = satoshi_per_bytes
+            .checked_mul_int(redeem_tx_size)
+            .ok_or(ArithmeticError::Overflow)?;
+        let amount = fee.try_into().map_err(|_| Error::<T>::TryIntoIntError)?;
+        Ok(Amount::new(amount, wrapped_currency))
+    }
+
     /// Get all vaults that:
     /// - are below the premium redeem threshold, and
     /// - have a non-zero amount of redeemable tokens, and thus
     /// - are not banned
     ///
-    /// Maybe returns a tuple of (VaultId, RedeemableTokens)
-    /// The redeemable tokens are the currently vault.issued_tokens - the vault.to_be_redeemed_tokens
-    pub fn get_premium_redeem_vaults() -> Result<Vec<(DefaultVaultId<T>, Amount<T>)>, DispatchError> {
-        let mut suitable_vaults = Vaults::<T>::iter()
-            .filter_map(|(vault_id, vault)| {
-                let rich_vault: RichVault<T> = vault.into();
+    /// Return a tuple of (VaultId, RedeemTokens to get `max_premium` from vault)
+    pub fn get_premium_redeem_vaults(
+        redeem_transaction_size: u32,
+    ) -> Result<Vec<(DefaultVaultId<T>, Amount<T>)>, DispatchError> {
+        let premium_reward_rate = ext::fee::premium_redeem_reward_rate::<T>();
 
-                let redeemable_tokens = rich_vault.redeemable_tokens().ok()?;
-
-                if !redeemable_tokens.is_zero() && Self::is_vault_below_premium_threshold(&vault_id).unwrap_or(false) {
-                    Some((vault_id, redeemable_tokens))
+        let mut suitable_vaults = Vaults::<T>::iter()
+            .filter_map(|(vault_id, _vault)| {
+                // The calculation to calculate redeem tokens to get `max_premium` is referenced from
+                // redeem pallet `request_redeem` method.
+                // BurnTokensForReachingPremiumThreshold = BurnWrap + InclusionFee
+                // RedeemTokens = BurnTokensForReachingPremiumThreshold / (1 - RedeemFee)
+                let max_premium_in_collateral = Self::get_vault_max_premium_redeem(&vault_id).ok()?;
+                let redeem_amount_wrapped_in_collateral =
+                    max_premium_in_collateral.checked_div(&premium_reward_rate).ok()?;
+                let burn_wrap = redeem_amount_wrapped_in_collateral
+                    .convert_to(vault_id.wrapped_currency())
+                    .ok()?;
+                let inclusion_fee =
+                    Self::calculate_inclusion_fee(vault_id.wrapped_currency(), redeem_transaction_size).ok()?;
+
+                let vault_to_burn_tokens = burn_wrap.checked_add(&inclusion_fee).ok()?;
+
+                let redeem_fee = ext::fee::get_redeem_fee_value::<T>();
+                let amount_wrapped = UnsignedFixedPoint::<T>::one().saturating_sub(redeem_fee);
+
+                let request_redeem_tokens_for_max_premium = vault_to_burn_tokens.checked_div(&amount_wrapped).ok()?;
+
+                if Self::ensure_not_banned(&vault_id).is_ok() && !request_redeem_tokens_for_max_premium.is_zero() {
```

Comment:
`will_be_below_premium_threshold` check method has been removed from here, but this generates a corner case since `vault_to_burn_tokens` will always be a nonzero amount, since `inclusion_fee` is a nonzero amount, therefore the rpc is giving out `max_premium` for vaults which are above the `PremiumRedeemThreshold`. 

To avoid this either in the if case `
- will_be_below_premium_threshold` method needs to be present OR
- check `max_premium_in_collateral` is non zero amount

parachain test case where the corner case can be trigger: `integration_test_try_get_premium_vaults_which_are_sufficiently_collateralized`

Reply:
added back the condition